### PR TITLE
chore: update lerna to 8.1.8 from 6.6.2

### PR DESCRIPTION
**Pull request by AtofStryker**
- Closes N/A

### Additional details

This PR updates `lerna` from `6.6.2` to `8.1.8`. The main reason for this update is some developers, including myself, is that there seems to be cache poisoning on the installl job which sometimes requires the use of `sudo`. This also seems to be happening in CI in some cases, like seen [in the "Install node modules"](https://app.circleci.com/pipelines/github/cypress-io/cypress/64791/workflows/dcb8f9b2-387d-49b9-ab78-bcac9da7e2cd/jobs/2692092) job here where `lerna` handes on @packages/server`. To get this working, I:

* installed `lerna@8.1.8`.
* ran `lerna repair`. This is the bulk of the changes in the `nx.json`, `lerna.json`, removal of `@nrwl/nx-cloud` package, `.gitignore` additions, and `nx` key being added to packages.
* I also needed to pin `typescript ` `5.4.5` since we can't cut over from `importsNotUsedAsValues`in certain packages yet (removed in 5.5 and 5.6.3 got installed and wasnt noticed because cache poisoning)
* I also updated the `sync-exported-npm-with-cli` to use the correct `npm-packlist` since we were explicitly referencing it but didn't have a version installed. The failure in the `@cypress/react18` build job went unnoticed due to caching the build since no source files changed. I pinned `npm-packlist` and made the updates on a61cee0b401a23a2beed7ddb4d65b0bef0e85ba2 (you can see [v9 npm-packlist](https://github.com/npm/npm-packlist/tree/v9.0.0) vs [v6 and older npm-packlist](https://github.com/npm/npm-packlist/tree/v6.0.0). my guess is this was updated passively since we didnt pin it and it never failed since the cache did not bust for `@cypress/react18` since we never made source file changes.

### Steps to test

everything should work as expected. this is just an update to our build/repo tooling.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: scripts/sync-exported-npm-with-cli.js [hunk 1]
```diff
@@ -16,66 +17,72 @@ const currentPackageDir = process.cwd()
...
-  // 2. Move all of the files that would be published under @cypress/react
+console.log(currentPackageDir)
+
+const arborist = new Arborist({ path: currentPackageDir })
+
+arborist.loadActual().then((tree) => {
+  packlist(tree)
+  .then((files) => {
+    // 2. Move all of the files that would be published under @cypress/react
   // to be copied under cli/react (drop the @cypress namespace)
```
Comment (mschile) [inline scripts/sync-exported-npm-with-cli.js]:
```suggestion
    // to be copied under cli/react (drop the @cypress namespace)
```

File: scripts/sync-exported-npm-with-cli.js [hunk 2]
```diff
@@ -16,66 +17,72 @@ const currentPackageDir = process.cwd()
...
+    const cliPackageConfig = require(path.join(cliPath, 'package.json'))
 
-  const subPackageExports = cliPackageConfig.exports[`./${exportName}`] = {}
-  const esmEntry = isModule ? currentPackageConfig.main : currentPackageConfig.module
+    const subPackageExports = cliPackageConfig.exports[`./${exportName}`] = {}
+    const esmEntry = isModule ? currentPackageConfig.main : currentPackageConfig.module
 
-  if (types) {
+    if (types) {
     // ./react/dist/cypress-react-cjs.js, etc
```
Comment (mschile) [inline scripts/sync-exported-npm-with-cli.js]:
```suggestion
      // ./react/dist/cypress-react-cjs.js, etc
```

File: scripts/sync-exported-npm-with-cli.js [hunk 3]
```diff
@@ -16,66 +17,72 @@ const currentPackageDir = process.cwd()
...
-    subPackageExports.types = `./${exportName}/${types}`
-  }
+      subPackageExports.types = `./${exportName}/${types}`
+    }
 
-  if (esmEntry) {
+    if (esmEntry) {
     // ./react/dist/cypress-react-esm.js, etc
```
Comment (mschile) [inline scripts/sync-exported-npm-with-cli.js]:
```suggestion
      // ./react/dist/cypress-react-esm.js, etc
```

File: scripts/sync-exported-npm-with-cli.js [hunk 4]
```diff
@@ -16,66 +17,72 @@ const currentPackageDir = process.cwd()
...
-    subPackageExports.import = `./${exportName}/${esmEntry}`
-  }
+      subPackageExports.import = `./${exportName}/${esmEntry}`
+    }
 
-  if (!isModule) {
+    if (!isModule) {
     // ./react/dist/cypress-react-cjs.js, etc
```
Comment (mschile) [inline scripts/sync-exported-npm-with-cli.js]:
```suggestion
      // ./react/dist/cypress-react-cjs.js, etc
```

File: scripts/sync-exported-npm-with-cli.js [hunk 5]
```diff
@@ -16,66 +17,72 @@ const currentPackageDir = process.cwd()
...
+    // Remove output directory to clean up old files before building
+    fs.removeSync(outDir)
 
-  // 3. For each file, mkdir if not exists, and then copy the dist'd assets over
-  // to write to the `cliPackageConfig` at the end
-  files.forEach((f) => {
+    // 3. For each file, mkdir if not exists, and then copy the dist'd assets over
+    // to write to the `cliPackageConfig` at the end
+    files.forEach((f) => {
     // mkdir if not exists
```
Comment (mschile) [inline scripts/sync-exported-npm-with-cli.js]:
```suggestion
      // mkdir if not exists
```

File: scripts/sync-exported-npm-with-cli.js (hunk 1 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: addressed in a3c79214c7

File: scripts/sync-exported-npm-with-cli.js (hunk 3 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: addressed in a3c79214c7

File: scripts/sync-exported-npm-with-cli.js (hunk 2 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: addressed in a3c79214c7

File: scripts/sync-exported-npm-with-cli.js (hunk 5 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: addressed in a3c79214c7

File: scripts/sync-exported-npm-with-cli.js (hunk 4 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: addressed in a3c79214c7

File: scripts/sync-exported-npm-with-cli.js [hunk 6]
```diff
@@ -16,66 +17,72 @@ const currentPackageDir = process.cwd()
 // 1. We'll run npm's own "packlist" against the npm package to be published (@cypress/react, etc)
 // to make sure we don't miss any files when we copy them over to the CLI package
 // The files that will be returned here are the ones from @cypress/react's package.json['files'] key.
-packlist({ path: currentPackageDir })
-.then((files) => {
-  // 2. Move all of the files that would be published under @cypress/react
+console.log(currentPackageDir)
```
Comment (mschile) [inline scripts/sync-exported-npm-with-cli.js]:
```suggestion
console.log('current package directory: ', currentPackageDir)
```

**Review by mschile — APPROVED** [review]

File: scripts/sync-exported-npm-with-cli.js (hunk 6 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: im just going to remove this as I think I added it due to debugging issues

Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js]: removed in 6b35b78201

File: nx.json [hunk 7]
```diff
@@ -54,5 +44,6 @@
       "!{projectRoot/cypress.config.{ts,js}",
       "!{projectRoot/**/*.cy.ts"
     ]
-  }
+  },
+  "nxCloudAccessToken": "ZmNlNjA0YzAtNTM1NS00MDIwLWFlMWItNWYxYzNiMjQ4N2VkfHJlYWQtb25seQ=="
```
Comment (jennifer-shehane) [inline nx.json:48]: @AtofStryker Should this be public? I know you just moved it...

File: scripts/sync-exported-npm-with-cli.js [hunk 8]
```diff
@@ -6,6 +6,7 @@
  * with this version of the binary
  */
 const path = require('path')
+const Arborist = require('@npmcli/arborist')
```
Comment (jennifer-shehane) [inline scripts/sync-exported-npm-with-cli.js:9]: cool package name

**Review by jennifer-shehane — APPROVED** [review]

File: scripts/sync-exported-npm-with-cli.js (hunk 8 above)
Comment (AtofStryker) [reply scripts/sync-exported-npm-with-cli.js:9]: 🌲

File: nx.json (hunk 7 above)
Comment (AtofStryker) [reply nx.json:48]: its not supposed to be as it's [legacy configuration](https://nx.dev/ci/recipes/security/access-tokens#legacy-methods-of-setting-ci-access-tokens) but I figured addressing that was out of scope here. At some point we should see what we can clean up in our monorepo tools/processes.

**Pull request merged**